holidays/countries/china.py

```diff
@@ -9,53 +9,375 @@
+            self._add_holiday(self[dt], dt + td(days=+1))
+            self._add_holiday(self[dt], dt + td(days=+2))
+            if self._year >= 2020 and dt == date(self._year, MAY, 1):
+                # 2 Extra Days for May Day >= 2020:
+                self._add_holiday(self[dt], dt + td(days=+3))
+                self._add_holiday(self[dt], dt + td(days=+4))
+        elif self._is_friday(dt):
+            # SAT-SUN
+            self._add_holiday(self[dt], dt + td(days=+1))
+            self._add_holiday(self[dt], dt + td(days=+2))
+            if self._year >= 2020 and dt == date(self._year, MAY, 1):
+                # 2 Extra Days for May Day >= 2020:
+                self._add_holiday(self[dt], dt + td(days=+3))
+                self._add_holiday(self[dt], dt + td(days=+4))
+        elif self._is_saturday(dt):
+            if self._year <= 2011 or self._year >= 2020:
+                # SUN-MON
+                self._add_holiday(self[dt], dt + td(days=+1))
+                self._add_holiday(self[dt], dt + td(days=+2))
+                if self._year >= 2020 and dt == date(self._year, MAY, 1):
+                    # 2 Extra Days for May Day >= 2020:
+                    self._add_holiday(self[dt], dt + td(days=+3))
+                    self._add_holiday(self[dt], dt + td(days=+4))
+            elif 2012 <= self._year <= 2019:
+                # [FRI]-SUN
+                self._add_holiday(self[dt], dt + td(days=-1))
+                self._add_holiday(self[dt], dt + td(days=+1))
+        elif self._is_sunday(dt):
+            if self._year <= 2007 or 2012 <= self._year <= 2019:
+                # MON-TUE
+                self._add_holiday(self[dt], dt + td(days=+1))
+                self._add_holiday(self[dt], dt + td(days=+2))
+            elif 2008 <= self._year <= 2011 or self._year >= 2020:
+                # [SAT]-MON
+                self._add_holiday(self[dt], dt + td(days=-1))
+                self._add_holiday(self[dt], dt + td(days=+1))
+                if self._year >= 2020 and dt == date(self._year, MAY, 1):
+                    # 2 Extra Days for May Day >= 2020:
+                    self._add_holiday(self[dt], dt + td(days=+2))
+                    self._add_holiday(self[dt], dt + td(days=+3))
+
+    def _populate_public_holidays(self):
+        # Proclamation of the People's Republic of China on Oct 1, 1949.
+        if self._year <= 1949:
+            return None
+
+        # 元旦 (simp.) / 新年 (trad.)
+        # Status: In-Use (Statutory).
+        # Jan 1 in 1949, 1999, 2007, and 2013 revision.
+        # Consecutive Holidays are available from 2002, except in 2014/2016/2017/2018.
+
+        # New Year's Day.
+        jan_1 = self._add_new_years_day(tr("元旦"))
+
+        if (
+            self.observed
+            and (2002 <= self._year <= 2015 or self._year >= 2020)
+            and self._year
+            not in {
+                2008,
+                2013,
+                2014,
+                2023,
```

Comment:
It'd be more compact this way:
```suggestion
                2023
```